### PR TITLE
Sanitize more config options and stack traces

### PR DESCRIPTION
**Pull request by MorrisJobke**
cc @nickvergessen @schiessle @rullzer @LukasReschke

### CODE REVIEW

File: lib/private/SystemConfig.php [hunk 1]
```diff
@@ -55,6 +55,12 @@ class SystemConfig {
 		'objectstore' => [
 			'arguments' => [
 				'password' => true,
+				'options' => [
+					'credentials' => [
+						'key' => true,
+						'secret' => true,
+					]
+				]
```
Comment (MorrisJobke) [inline lib/private/SystemConfig.php]: Allows to be configuration compatible with https://doc.owncloud.com/server/9.0/admin_manual/enterprise_external_storage/s3_swift_as_primary_object_store_configuration.html#amazon-s3

**Comment by icewind1991** [comment]
:+1:

**Comment by nickvergessen** [comment]
:+1:

**Comment by nickvergessen** [comment]
Backport?

**Comment by MorrisJobke** [comment]
> Backport?

This is not available in stable9 ;)

**Pull request merged**